pkg/process/util/api/payload.go

```diff
@@ -0,0 +1,43 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package api
+
+import (
+	"fmt"
+	"strconv"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+)
+
+var (
+	tlmBytesIn = telemetry.NewCounter("process", "payloads_bytes_in",
+		[]string{"type"}, "Count of bytes before encoding payload")
+	tlmBytesOut = telemetry.NewCounter("process", "payloads_bytes_out",
+		[]string{"type"}, "Count of bytes after encoding payload")
+)
+
+// EncodePayload encodes a process message into a payload
+func EncodePayload(m model.MessageBody) ([]byte, error) {
+	msgType, err := model.DetectMessageType(m)
+	if err != nil {
+		return nil, fmt.Errorf("unable to detect message type: %s", err)
+	}
+
+	typeTag := "type:" + strconv.Itoa(int(msgType))
```

Comment:
might be worth to convert back this to a string 🤔 or we can learn https://github.com/DataDog/agent-payload/blob/6a8609f0fa8fb7ec6c5a9142bd459e1635e316bb/process/message.go#L75-L86 by heart 😬